### PR TITLE
Nisp 1754 Fixing chart bug when small amount is displayed

### PR DESCRIPTION
**Pull request by roundrobin99**
There are a few places for this fix to work.

AccountController : Fix to display a minimum width of 29%
nisp.scss : Added a reduced font size and line height so the text is nice and centered on mobile devices

### CODE REVIEW

**Comment by roundrobin99** [comment]
Used minimum width of 29 as this is the minimum needed when a variety of numbers will be displayed (i.e 40.11, 40.88, 40.99).

**Comment by roundrobin99** [comment]
To test you will need to use NISP-STUB using the NISP-1754 branch.

**Comment by roundrobin99** [comment]
iphone_6_breakdown_fixed
desktop_breakdown_fixed

**Comment by dspork** [comment]
Is the ideal solution? It may work in this case but not all. What if the second bar was much higher?

**Comment by roundrobin99** [comment]
What do you mean ? Can you provide an example ?

File: app/uk/gov/hmrc/nisp/controllers/AccountController.scala [hunk 1]
```diff
@@ -105,7 +105,13 @@ trait AccountController extends NispFrontendController with AuthorisedForNisp wi
   def calculateChartWidths(currentAmountModel: SPAmountModel, forecastAmountModel: SPAmountModel): (SPChartModel, SPChartModel) = {
     // scalastyle:off magic.number
     if (forecastAmountModel.week > currentAmountModel.week) {
-      val currentChart = SPChartModel((currentAmountModel.week/forecastAmountModel.week * 100).toInt, currentAmountModel)
+      val currentPercentage = (currentAmountModel.week/forecastAmountModel.week * 100).toInt     
+      val currentChart = if (currentPercentage<29) {
```
Comment (dspork) [inline app/uk/gov/hmrc/nisp/controllers/AccountController.scala]: We can use `.max(29)` here to avoid the if statement.

**Comment by dspork** [comment]
If say, the forecast was instead £239,  the chart should standing at 17%.  Yes your fix will probably will the bar from shrinking, but it will look very disproportionate?

Has @steven-borthwick agreed to this, as well as the font-size change?

**Comment by steven-borthwick** [comment]
Hi

We need a proper fix, this was just to make it readable. Let's go through
it next week.

Steve

On Friday, 18 March 2016, Callum Turnbull <notifications@github.com> wrote:

> If say, the forecast was instead £239, the chart should standing at 17%.
> Yes your fix will probably will the bar from shrinking, but it will look
> very disproportionate?
>
> Has @steven-borthwick <https://github.com/steven-borthwick> agreed to
> this, as well as the font-size change?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/hmrc/nisp-frontend/pull/83#issuecomment-198402421>
>


-- 
Steven Borthwick | Interaction Designer | NI & State Pensions | Department
for Work and Pensions | Newcastle upon Tyne | 07870 469540

**Comment by dspork** [comment]
Ok, I'll close this for now, we can talk about a proper solution on Monday (I'll re-open if necessary)

File: app/uk/gov/hmrc/nisp/controllers/AccountController.scala (hunk 1 above)
Comment (roundrobin99) [inline app/uk/gov/hmrc/nisp/controllers/AccountController.scala]: Cool.

**Comment by roundrobin99** [comment]
We need to think what will be the odds of this happening. 
How likely is it to have an amount of £239 when current amount is £40 ?

**Pull request closed without merging**